File: SimpleRecycleView/app/src/main/java/com/pavel/dolbik/simplerecycleview/MainActivity.java
package com.pavel.dolbik.simplerecycleview;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements Adapter.OnItemClickListener{


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
        Adapter adapter = new Adapter(getData());
        adapter.setOnItemClickListener(this);

        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.list);
        recyclerView.setHasFixedSize(true);
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(layoutManager);
    }


    private String[] getData() {
        String[] data = new String[100];
        for (int i = 0; i < 100; i++) {
            data[i] = "item "+i;
        }
        return data;
    }


    @Override
    public void onItemClick(String item) {
        Toast.makeText(this, "Item - "+item, Toast.LENGTH_SHORT).show();
    }
}

File: KotlinAdapter/app/src/main/java/com/dolbik/pavel/kotlinadapter/MainActivity.kt
package com.dolbik.pavel.kotlinadapter

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import com.arellomobile.mvp.MvpAppCompatActivity
import com.arellomobile.mvp.presenter.InjectPresenter
import com.dolbik.pavel.kotlinadapter.adapters.MainAdapter
import com.dolbik.pavel.kotlinadapter.common.InfiniteScrollListener
import com.dolbik.pavel.kotlinadapter.common.NewsItem
import com.dolbik.pavel.kotlinadapter.mvp.MainPresenter
import com.dolbik.pavel.kotlinadapter.mvp.MainView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : MvpAppCompatActivity(), MainView {

    @InjectPresenter
    lateinit var presenter: MainPresenter


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        list.apply {
            val linearLayout = LinearLayoutManager(context)
            setHasFixedSize(true)
            layoutManager = linearLayout
            adapter = MainAdapter()
            clearOnScrollListeners()
            addOnScrollListener(InfiniteScrollListener({ presenter.requestNews() }, linearLayout))
        }
    }


    override fun setData(data: List<NewsItem>) {
        (list.adapter as MainAdapter).addNews(data)
    }
}
